### PR TITLE
Poly-fungible v2 migration

### PR DESCRIPTION
**Pull request by sirlensalot**
Makes following changes:
- introduce draft of `poly-fungible-v2`, which incorporates manifests instead of just URI
- organize namespaces: `kip` for standards, `util` for fungible-util
- conform `hft` to `poly-fungible-v2`
- code organization

### CODE REVIEW

File: pact/hft.pact [hunk 1]
```diff
@@ -111,52 +65,63 @@
       newbal)
   )
 
-  (defcap ROTATE (token:string account:string)
+  (defcap SUPPLY:bool (id:string supply:decimal)
```
Comment (ggobugi27) [inline pact/hft.pact:68]: Should this replace `UPDATE_SUPPLY`?

File: pact/kip/poly-fungible-v2.pact [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      \ must match existing guard. If RECEIVER does not exist, account is created."
+    @model
+      [ (property (> amount 0.0))
+        (property (!= id ""))
+        (property (!= sender ""))
+        (property (!= receiver ""))
+        (property (!= target-chain ""))
+      ]
+  )
+
```
Comment (ggobugi27) [inline pact/kip/poly-fungible-v2.pact:165]: Add `update-supply`?

**Review by ggobugi27 — COMMENTED** [review]
The changes with `poly-fungible-v2` looks all good, but not clear about the new defcap `SUPPLY`, if it's replacing `UPDATE_SUPPY` or separate event-emitting defcap. 
Also opened a #13 to add `get-ledger-key` and tests.

**Comment by buckie** [comment]
This looks reasonable to me

File: pact/hft.pact (hunk 1 above)
Comment (sirlensalot) [reply pact/hft.pact:68]: No, because `update-supply` computes the amount and `UPDATE_SUPPLY` is the private cap to guard it.

Comment (sirlensalot) [reply pact/hft.pact:68]: However this caught a bug which is that `update-supply` is not emitting `SUPPLY`, so I've addressed that. 👍

File: pact/kip/poly-fungible-v2.pact (hunk 2 above)
Comment (sirlensalot) [reply pact/kip/poly-fungible-v2.pact:165]: No b/c that is a private/implementation-specific function.

File: pact/hft.pact [hunk 3]
```diff
@@ -165,194 +134,194 @@
...
-          (credit token receiver g amount))
-      )
+  (defun mint:string
+    ( id:string
+      account:string
+      guard:guard
+      amount:decimal
     )
+    (with-capability (MINT id account amount)
+      (with-capability (CREDIT id account)
```
Comment (ggobugi27) [inline pact/hft.pact:230]: This line is probably unnecessary?

File: pact/hft.pact [hunk 4]
```diff
@@ -165,194 +134,194 @@
...
-      (with-capability (MINT token account amount)
-        (with-capability (CREDIT token account)
-          (credit token account guard amount)))
+  (defun burn:string
+    ( id:string
+      account:string
+      amount:decimal
     )
+    (with-capability (BURN id account amount)
+      (with-capability (DEBIT id account)
```
Comment (ggobugi27) [inline pact/hft.pact:241]: Same with above comment in `mint`. Unnecessary because DEBIT is composed in BURN.

**Review by ggobugi27 — APPROVED** [review]

**Review by ggobugi27 — APPROVED** [review]

**Pull request merged**